plugins/package-managers/cocoapods/src/main/kotlin/CocoaPods.kt

```diff
@@ -229,13 +229,16 @@ private const val LOCKFILE_FILENAME = "Podfile.lock"
 
 private const val SCOPE_NAME = "dependencies"
 
-private val NAME_AND_VERSION_REGEX = "(\\S+)\\s+(.*)".toRegex()
-
 private fun getPackageReferences(podfileLock: File): Set<PackageReference> {
     val versionForName = mutableMapOf<String, String>()
     val dependenciesForName = mutableMapOf<String, MutableSet<String>>()
     val root = yamlMapper.readTree(podfileLock)
 
+    fun parseNameAndVersion(entry: String): Pair<String, String?> {
```

Comment:
This function does not use anything from this scope. Could you please move it out, so the code of `getPackageReferences()` is mixing less abstraction levels?